### PR TITLE
Rename endpoint folder -> module folder and add core folder 

### PR DESCRIPTION
**Pull request by maingockien01**

### CODE REVIEW

File: API/src/core/ApiError.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    ) {
+        super(type);
+    }
+
+    public handle(res: Response): Response {
+        return this.response.send(res);
+    }
+}
+
+export class BadTokenError extends ApiError {
```
Comment (jeff-knurek) [inline API/src/core/ApiError.ts:36]: we're not going to be adding auth to the api for a while I expect, so maybe it's better to leave the token errors out for now to not add unused code

File: API/src/core/ApiResponse.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import { Response } from 'express';
+
+enum StatusCode {
```
Comment (jeff-knurek) [inline API/src/core/ApiResponse.ts:3]: what's the use of StatusCode? 
I think for anything that isn't 200, the response code and message should be enough

File: API/src/core/ApiResponse.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    /*
+     * @function        sanitize
+     * @desc            prepare json object for the response
+     *                  remove all unneeded fields and functions
+     * @params          T {extends ApiResponse} subclass of ApiResponse contatining data for response.
+     *
+     * @returns         an object contains: statusCode, data (if subclass has the field)
+     */
+    private static sanitize<T extends ApiResponse>(response: T): T {
```
Comment (jeff-knurek) [inline API/src/core/ApiResponse.ts:61]: one thing to add (or at least think about), is that the API will take a global parameter of the response format to return (ie json, csv, text,.....) so having something that processes the different format types in the response will be needed

**Review by jeff-knurek — COMMENTED** [review]
the PR doesn't show that the `src/endpoints` directory is being removed _(meaning there is now duplicate code, not moved code)_

File: API/src/core/ApiResponse.ts (hunk 3 above)
Comment (maingockien01) [reply API/src/core/ApiResponse.ts:61]: okay, I see. I didnt think about other format for the response. I'll change it.

File: API/src/core/ApiResponse.ts (hunk 2 above)
Comment (maingockien01) [reply API/src/core/ApiResponse.ts:3]: yeah, I think that status code is unnecessary actually.

**Pull request closed without merging**